client/landing/gutenboarding/components/header/index.tsx

```diff
@@ -179,3 +178,18 @@ const Header: FunctionComponent< Props > = ( { prev } ) => {
 };
 
 export default Header;
+
+/* eslint-disable */
+const BackButton = withI18n(
+	class InnerBackButton extends React.PureComponent< I18nProps & { prev?: string } > {
+		render() {
+			const { prev, __ } = this.props;
```

Comment:
Did you mention you had some issues getting the i18n updates through the `PureComponent` barrier 🙂 

One reason for that is that `__` is always the same object, no matter if the underlying `i18n` object got updated or not.

Usually, the changed `i18nLocale` prop should be enough to force rerender of a `PureComponent`, but sometimes (especially in Calypso), the translations can get updated with `i18nLocale` unchanged. E.g., in the community translator.

A good technique to solve that is to not return the raw `__` function from `makeI18n`, but `bind` it to create a new function with unique identity.

https://github.com/Automattic/wp-calypso/blob/master/packages/i18n-calypso/src/use-translate.js#L8-L10

Reply:
The reason I had difficulty with that was caused by the issue you noted in your review here:

https://github.com/Automattic/wp-calypso/pull/39480#discussion_r380139496

As long as locale data is set before the first render, `PureComponent`s are happy. Subsequent updates to the `i18nLocale` prop trigger the desired rerenders. That was my motivation for adding the `i18nLocale` prop.

> A good technique to solve that is to not return the raw `__` function from `makeI18n`, but `bind` it to create a new function with unique identity.

I considered this, but preferred relying on `i18nLocale` changes. The issue should be fixed by the changes in this PR.